### DISCUSSION TITLE
Single session / device per user

### DISCUSSION THREAD

**Discussion by NixBiks** [supabase/supabase#5425] [Questions]
Is it possible to restrict users to a single session only. If a user signs in on device A while the user is already signed in on device B then the user should be signed out from device B. Is that possible somehow?

**Comment by NixBiks** [comment]
No suggestions available on this one?

**Comment by GaryAustin1** [comment]
A bullet proof way is hard to do with JWT's which can't easily be expired before their time and general supabase/gotrue signin process.  There is no built in server support to detect signin or singout someone.  You can use triggers on auth.users to detect signin, but have no way to stop a currently active token set.

But...
One thing close depending, on level of security versus or just being a deterrent, might be a "session" ID table with one entry per userID.  This session ID is set on signin and compared in RLS.  Any requests from another client get denied.  

You generate a random ID and set a custom session header when you init the supabase client.
As part of sign in process you set the session table with this value with RLS on user_ID.
Then on all requests you RLS check this header against the session table with a function like auth.uid() to get header info from postgrest and compare against the session table.
The other client will get RLS rejected on sign in from another account.

https://github.com/supabase/supabase/discussions/4755 discusses some of the concepts.
https://postgrest.org/en/stable/api.html#http-context  PostgREST http info

A determined user could of course exchange the ID and hack the js code with a shared id.  This could be made tougher by checking time setting of last session table update and having the client update this every hour or so to  keep it valid.  I believe you can re run supabase client init with a new header and not lose your signed in user, but you would need to verify that.
There also might be some info in the request headers that is somewhat unique to different clients, or real session info that could be paired with the id sent from the client.

**Comment by NehalDamania** [comment]
Hi, would like to know, is this not supported out of the box yet?

**Reply by GaryAustin1** [reply]
https://supabase.com/docs/guides/auth/sessions#limiting-session-lifetime-and-number-of-allowed-sessions-per-user
For Pro.
